Cargo.toml

```diff
@@ -214,6 +214,7 @@ generic-array = { version = "0.14.7", default-features = false }
 gethostname = "0.2.3"
 getrandom = "0.2.10"
 goauth = "0.13.1"
+governor = "0.6.3"
```

Comment:
introducing a new dependency will be a hard no on backport. is it really necessary? we already have the stream rate limiter logic that could be reused instead

Reply:
you are missing the point. it's (probably) fine for master. we are not taking a new dependency to the stable branch

Reply:
Why is it a hard no for backport?

Reply:
it's not tested. how hard is this guys? do we really all need first hand experience crashing mb to understand the gravity of what we're doing here?

Reply:
I have changed the implementation to use a very simple rate limiter without using governor for back port purpose.